File: Plugin 1/Class1.vb
Public Class Class1
    Implements PluginSample.Interfaces.IPlugin

    Private objHost As PluginSample.Interfaces.IHost

    Public Sub Initialize(ByVal Host As PluginSample.Interfaces.IHost) Implements PluginSample.Interfaces.IPlugin.Initialize
        objHost = Host
    End Sub

    Public ReadOnly Property Name() As String Implements PluginSample.Interfaces.IPlugin.Name
        Get
            Return "Addition (VB.NET)"
        End Get
    End Property

    Public Function Calculate(ByVal int1 As Integer, ByVal int2 As Integer) As Double Implements PluginSample.Interfaces.IPlugin.Calculate
        Return int1 + int2
    End Function
End Class

File: Plugin 3/Class1.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using PluginSample.Interfaces;

namespace Plugin_3
{
  public class Class1 : IPlugin
  {
    private IHost _host;

    public void Initialize(IHost host)
    {
      _host = host;
    }

    public string Name
    {
      get { return "Subtraction (C#)"; }
    }

    public Double Calculate(int x, int y)
    {
      return x - y;
    }
  }
}
